src/sql/workbench/contrib/notebook/test/browser/dataResourceDataProvider.test.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as TypeMoq from 'typemoq';
+import * as azdata from 'azdata';
+import * as os from 'os';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as uuid from 'uuid';
+import * as sinon from 'sinon';
+import { DataResourceDataProvider } from '../../browser/outputs/gridOutput.component';
+import { IDataResource } from 'sql/workbench/services/notebook/browser/sql/sqlSessionManager';
+import { ResultSetSummary } from 'sql/workbench/services/query/common/query';
+import { TestNotificationService } from 'vs/platform/notification/test/common/testNotificationService';
+import { TestFileDialogService, TestEditorService } from 'vs/workbench/test/browser/workbenchTestServices';
+import { SerializationService } from 'sql/platform/serialization/common/serializationService';
+import { SaveFormat, ResultSerializer } from 'sql/workbench/services/query/common/resultSerializer';
+import { InstantiationService } from 'vs/platform/instantiation/common/instantiationService';
+import { URI } from 'vs/base/common/uri';
+
+export class TestSerializationProvider implements azdata.SerializationProvider {
+	providerId: string;
+	constructor(providerId: string = 'providerId') { }
+
+	// Write data to file
+	startSerialization(requestParams: azdata.SerializeDataStartRequestParams): Thenable<azdata.SerializeDataResult> {
+		let data: string = '';
+		requestParams.rows.forEach((row) => {
+			row.forEach((element) => {
+				data += element.displayValue + ' ';
+			});
+			data += '\n';
+		});
+		fs.writeFile(requestParams.filePath, data);
```

Comment:
This is async so you open the test up to race condition failures here. Use an awaitable version instead

Reply:
should I be using fs.writeFileSync? I'm getting a warning for unexpected sync method

Reply:
No, we should never be using the sync methods. They block the entire JS thread while running so are best avoided.

Use the async methods from fs.promises instead and just await on them to achieve the same thing while still allowing the thread to complete other tasks while this is running.